services/http.go

```diff
@@ -0,0 +1,58 @@
+package services
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/getlantern/golog"
+
+	"github.com/getlantern/flashlight/v7/common"
+)
+
+// it is the caller's responsibility to read the response body to completion and close it
+func post(
+	originURL string,
+	buf io.Reader,
+	rt http.RoundTripper,
+	user common.UserConfig,
+	logger golog.Logger,
+) (io.ReadCloser, int64, error) {
+	req, err := http.NewRequest("POST", originURL, buf)
+	if err != nil {
+		return nil, 0, fmt.Errorf("unable to create request for %s: %w", originURL, err)
+	}
+
+	common.AddCommonHeaders(user, req)
+	req.Header.Set("Content-Type", "application/x-protobuf")
+	req.Header.Set("Accept", "application/x-gzip")
```

Comment:
I see this is where the gzip assumption starts.  Again, does this back-end respect this header for "application/x-protobuf"?  Protocol buffer messages are usually dense, so I don't think it's typical to apply compression.

Reply:
Again, I just assume it's compressed. I'll look at `lantern-cloud` and check if it is.

Reply:
Removing gzip compression.